perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,773 @@
+flags.DEFINE_integer('sysbench_warmup_seconds', 120,
+                     'The duration of the warmup run in which results are '
+                     'discarded.')
+flags.DEFINE_integer('sysbench_run_seconds', 480,
+                     'The duration of the actual run in which results are '
+                     'collected.')
+flags.DEFINE_integer('sysbench_thread_count', 16,
+                     'The number of test threads on the client side.')
+flags.DEFINE_integer('sysbench_latency_percentile', 99,
+                     'The latency percentile we ask sysbench to compute.')
+flags.DEFINE_integer('sysbench_report_interval', 2,
+                     'The interval we ask sysbench to report results.')
+
+BENCHMARK_INFO = {'name': 'mysql_service',
+                  'description': 'MySQL service benchmarks.',
+                  'scratch_disk': False,
+                  'num_machines': 1}
+
+# Query DB creation status once every 15 seconds
+DB_STATUS_QUERY_INTERVAL = 15
+
+# How many times we will wait for the service to create the DB
+# total wait time is therefore: "query interval * query limit"
+DB_STATUS_QUERY_LIMIT = 200
+
+AWS_PREFIX = ['aws', '--output=json']
+# Map from FLAGs.db_instance_cores to RDS DB Type
+RDS_CORE_TO_DB_CLASS_MAP = {
+    '1': 'db.m3.medium',
+    '4': 'db.m3.xlarge',
+    '8': 'db.m3.2xlarge',
+    '16': 'db.r3.4xlarge',  # m3 series doesn't have 16 core.
+}
+
+RDS_DB_ENGINE = 'MySQL'
+RDS_DB_ENGINE_VERSION = '5.6.23'
+RDS_DB_STORAGE_TYPE_GP2 = 'gp2'
+# Storage IOPS capacity of the DB instance.
+# Currently this is fixed because the cloud provider GCP does not support
+# changing this setting. As soon as it supports changing the storage size, we
+# will expose a flag here to allow caller to select a storage size.
+# Default GCP storage size is 1TB PD-SSD which supports 10K Read or 15K Write
+# IOPS (12.5K mixed).
+# To support 12.5K IOPS on EBS-GP, we need 4170 GB disk.
+RDS_DB_STORAGE_GP2_SIZE = '4170'
+
+# Constants defined for Sysbench tests.
+RAND_INIT_ON = 'on'
+DISABLE = 'disable'
+UNIFORM = 'uniform'
+OFF = 'off'
+MYSQL_ROOT_USER = 'root'
+MYSQL_ROOT_PASSWORD = 'Perfkit8'
+
+PREPARE_SCRIPT_PATH = '/usr/share/doc/sysbench/tests/db/parallel_prepare.lua'
+OLTP_SCRIPT_PATH = '/usr/share/doc/sysbench/tests/db/oltp.lua'
+
+SYSBENCH_RESULT_NAME_DATA_LOAD = 'sysbench data load time'
+SYSBENCH_RESULT_NAME_TPS = 'sysbench tps'
+SYSBENCH_RESULT_NAME_LATENCY = 'sysbench latency'
+NA_UNIT = 'NA'
+SECONDS_UNIT = 'seconds'
+MS_UNIT = 'milliseconds'
+
+# These are the constants that should be specified in GCP's cloud SQL command.
+DEFAULT_BACKUP_START_TIME = '07:00'
+GCP_MY_SQL_VERSION = 'MYSQL_5_6'
+GCP_PRICING_PLAN = 'PACKAGE'
+
+PERCENTILES_LIST = [1, 5, 50, 90, 99, 99.9]
+RESPONSE_TIME_TOKENS = ['min', 'avg', 'max', 'percentile']
+
+
+def GetInfo():
+  return BENCHMARK_INFO
+
+
+class DBStatusQueryError(Exception):
+    pass
+
+
+def _PercentileCalculator(numbers):
```

Comment:
seems like a good function to have in the util file.  I've seen similar code in other places and since this is more generic lets move it there.

numpy by the way can do this ... but likely we dont want the dependency.